configs/projects/agent-runtime.rb

```diff
@@ -0,0 +1,69 @@
+project 'agent-runtime' do |proj|
+  platform = proj.get_platform
+  # TODO: Is this an acceptable way to inherit from multiple possible agent version branches?
+  proj.setting(:agent_branch, ENV['AGENT_BRANCH'] || 'master')
```

Comment:
Hm. Originally I was definitely thinking it would be better to have a project per branch, but if there aren't many differences that might be a lot of overhead.

The main complication with this is going to be making sure we have a good naming/versioning scheme so that it's very obvious which runtime artifacts to be pulling in for each of the agent builds. The initial naming scheme is in https://tickets.puppetlabs.com/browse/VANAGON-98, but since you're adding the first / possibly only project with branches maybe it would make more sense to have something like agent-runtime-<branch>-<version>.

Reply:
I think I will go ahead and make separate project configs for each branch -- seems like as I add more cases to this file the difficulty of reading it could surpass the convenience of a single config.

Reply:
That makes sense. If you find you have a lot of duplicated code can always go for something like what we do in pl-build-tools: https://github.com/puppetlabs/pl-build-tools-vanagon/blob/master/configs/projects/pl-cmake.rb#L2-L3